common/build/eslint-plugin-fluid/src/test/tsconfig.json

```diff
@@ -1,8 +1,12 @@
 {
 	"compilerOptions": {
-		"rootDir": "./mockFiles",
-		"target": "ES5",
+		"rootDir": "example",
+		"target": "ESNext",
 		"composite": true,
+		"moduleResolution": "nodenext",
+		"module": "nodenext",
+		"allowImportingTsExtensions": true,
+		"noEmit": true
 	},
```

Comment:
Now with the `strict` rule, the mock test file will throw errors since some of the properties are declared in a class without providing an initializer and without assigning them a value in the constructor.

You can directly initialize the value like `signature: string = "signature"` or try like below (and this applies to `mockClassExpression.ts` too):

`mockClassDeclaration.ts`

```ts
/*!
 * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
 * Licensed under the MIT License.
 */

/**
 * @public
 */
class MockClass {
	/**
	 * @internal
	 */
	invalidInternal(): void {}

	/**
	 * @alpha
	 */
	invalidAlpha(): void {}

	/**
	 * @beta
	 */
	invalidBeta(): void {}

	/**
	 * @public
	 */
	invalidPublic(): void {}

	//@public
	invalidLineComment(): void {}

	/**
	 * Correctly implemented method with valid comment.
	 */
	validBlockComment(): void {}

	// Correctly implemented method with a line comment.
	validLineComment(): void {}

	validNoComment(): void {}

	signature: string;

	private _value = 1;

	/**
	 * @public
	 */
	public get value(): number {
		return this._value;
	}

	/**
	 * @public
	 */
	constructor() {
		this.signature = "signature";
	}
}

/**
 * @public
 *
 * Second class added to verify that the linter is able to parse multiple classes in a single file.
 */
class MockClassTwo {
	/**
	 * @internal
	 */
	invalidInternalTwo(): void {}

	// Valid property signature.
	validSignature: void;

	constructor() {
		this.validSignature = null;
	}
}
```